src/Database/Type/TimeType.php

```diff
@@ -105,42 +107,72 @@ public function marshal(mixed $value): Chronos|DateTimeInterface|null
             $value['microsecond']
         );
 
-        return new $class($format);
+        return new $this->_className($format);
     }
 
     /**
-     * @param string $value
-     * @return \Cake\I18n\DateTime|null
+     * @inheritDoc
      */
-    protected function _parseLocaleTimeValue(string $value): ?DateTime
+    public function manyToPHP(array $values, array $fields, Driver $driver): array
     {
-        /** @var class-string<\Cake\I18n\DateTime> $class */
-        $class = $this->_className;
+        foreach ($fields as $field) {
+            if (!isset($values[$field])) {
+                continue;
+            }
 
-        return $class::parseTime($value, $this->_localeMarshalFormat);
+            $value = $values[$field];
+            $instance = new $this->_className($value);
+            $values[$field] = $instance;
+        }
+
+        return $values;
+    }
+
+    /**
+     * Convert time data into the database time format.
+     *
+     * @param mixed $value The value to convert.
+     * @param \Cake\Database\Driver $driver The driver instance to convert with.
+     * @return mixed
+     */
+    public function toDatabase(mixed $value, Driver $driver): mixed
+    {
+        if ($value === null || is_string($value)) {
+            return $value;
+        }
+
+        return $value->format($this->_format);
+    }
+
+    /**
+     * Convert time values to PHP time instances
+     *
+     * @param mixed $value The value to convert.
+     * @param \Cake\Database\Driver $driver The driver instance to convert with.
+     * @return mixed
+     */
+    public function toPHP(mixed $value, Driver $driver): mixed
+    {
+        if ($value === null) {
+            return null;
+        }
+
+        return new $this->_className($value);
     }
 
     /**
      * Converts a string into a DateTime object after parsing it using the
      * formats in `_marshalFormats`.
      *
      * @param string $value The value to parse and convert to an object.
-     * @return \Cake\I18n\DateTime|\DateTimeImmutable|null
+     * @return \Cake\I18n\Time|null
      */
-    protected function _parseTimeValue(string $value): DateTime|DateTimeImmutable|null
+    protected function _parseTimeValue(string $value): ?Time
     {
-        $class = $this->_className;
         foreach ($this->_marshalFormats as $format) {
-            try {
-                $dateTime = $class::createFromFormat($format, $value);
-                // Check for false in case DateTime is used directly
-                if ($dateTime !== false) {
-                    return $dateTime;
-                }
-            } catch (InvalidArgumentException) {
-                // Chronos wraps DateTime::createFromFormat and throws
-                // exception if parse fails.
-                continue;
+            $time = $this->_className::parseTime($value, $format);
```

Comment:
Should we support `createFromFormat()` in Time so the same foramt specifiers are used across DateTime, Date and Time?

Reply:
Aren't Date and DateTimeType all using icu formats?

Reply:
Nope. They're using createFromFormat() (php DateTime) specifiers.

Reply:
I already added either `ICU Time Format` or `PHP Time Format` to the properties used in the TimeType format to make it clear which format is which. But this could be something we could add to other parts of the framework as well to make it a bit more clearer.

Reply:
Having some type classes using icu and others using php date formats is a little inconsistent.